src/formTimingTracking.js

```diff
@@ -10,30 +10,71 @@
     const getFieldName = (field) => field.getAttribute(fieldNameAttribute);
 
     const getFormName = (field) => field.closest("form").getAttribute(formNameAttribute);
+  
+    const trackFormSubmit = (e) => {
+        let formName = getFormName(e.target);
+        let fieldName = "Click";
+        let interactionType = "Submit";
+        subscribe({ formName, fieldName, interactionType });  
+    };
 
     const trackFormFieldEntry = (e) => {
         let fieldName = getFieldName(e.target);
         fieldsTimings[fieldName] = new Date().getTime();
     };
     
     const trackFormFieldLeave = (e) => {
-        let leaveType = e.type;
+        let interactionType = e.type;
         let formName = getFormName(e.target);
         let fieldName = getFieldName(e.target);
         // eslint-disable-next-line no-prototype-builtins
         if (fieldsTimings.hasOwnProperty(fieldName)) {
             let timeSpent = new Date().getTime() - fieldsTimings[fieldName];
             if (timeSpent > 0 && timeSpent < 1800000) {
-                subscribe({ formName, fieldName, leaveType, timeSpent });
+                subscribe({ formName, fieldName, interactionType, timeSpent });
             }
             delete fieldsTimings[fieldName];
         }
     };
 
-    document.querySelectorAll('input,select,textarea,datalist').forEach((elem) => {
+    const waitForElement = function(selector){
+            return new Promise(resolve => {
+                if (document.querySelector(selector)) {
+                    return resolve(document.querySelector(selector));
+                }
+
+                const observer = new MutationObserver(mutations => {
+                    if (document.querySelector(selector)) {
+                        resolve(document.querySelector(selector));
+                        observer.disconnect();
+                    }
+                });
+
+                observer.observe(document.body, {
+                    childList: true,
+                    subtree: true
+                });
+            });
+        };   
+
+    const addNewListeners = function() {
+        document.querySelectorAll('form').forEach((el) => {
+            el.addEventListener('submit', trackFormSubmit);
+        });
+        document.querySelectorAll('input,select,textarea,datalist').forEach((elem) => {
             elem.addEventListener('focus', trackFormFieldEntry);
             elem.addEventListener('change', trackFormFieldLeave);
             elem.addEventListener('blur', trackFormFieldLeave);
         });
+    };        
+
+    if(document.forms.length){ 
+      addNewListeners();
+    }
+    else{
+        waitForElement('form').then(elm => {
```

Comment:
`elm` is not used, should be removed.

Reply:
@rezik007 yes, it has been tested on our new demo - banking.demo.piwikpro.com